analysis/model/make_model_input.R

```diff
@@ -94,450 +108,485 @@ for (i in 1:nrow(active_analyses)) {
-    
-    df <- input[input$sub_bin_covid19_confirmed_history==FALSE & 
-                  input$cov_cat_sex=="Male" & 
-                  input$cov_num_age>=40 &
-                  input$cov_num_age<60,]
-    
-    df[,c(colnames(df)[grepl("sub_",colnames(df))],"cov_cat_sex")] <- NULL
+
+  # Make model input: prior cognitive impairment false -------------------------
+
+  if (active_analyses$analysis[i]=="sub_prior_cognitive_impairment_false") {
+
+    print('Make model input: sub_prior_cognitive_impairment_false')
+
+    df <- input[input$sub_bin_covid19_confirmed_history==FALSE &
+                  input$cov_bin_history_cog_imp_sympt == FALSE,]
+
+    df[,c(colnames(df)[grepl("sub_",colnames(df))],"cov_bin_history_cog_imp_sympt")] <- NULL
     
     check_vitals(df)
     readr::write_rds(df, file.path("output", paste0("model_input-",active_analyses$name[i],".rds")), compress = "gz")
     print(paste0("Saved: output/model_input-",active_analyses$name[i],".rds"))
     rm(df)
-    
+
   }
-  
-  # Make model input: sub_male_age_60_79 -----------------------------------------
-  
-  if (active_analyses$analysis[i]=="sub_male_age_60_79") {
-    
-    print('Make model input: sub_male_age_60_79')
-    
-    df <- input[input$sub_bin_covid19_confirmed_history==FALSE & 
-                  input$cov_cat_sex=="Male" & 
-                  input$cov_num_age>=60 &
-                  input$cov_num_age<80,]
-    
-    df[,c(colnames(df)[grepl("sub_",colnames(df))],"cov_cat_sex")] <- NULL
+
+  # Make model input: prior parkinson true -------------------------------------
+
+  if (active_analyses$analysis[i]=="sub_prior_parkinson_true") {
+
+    print('Make model input: sub_prior_parkinson_true')
+
+    df <- input[input$sub_bin_covid19_confirmed_history==FALSE &
+                  input$cov_bin_history_any_dementia == TRUE,]
+
+    df[,c(colnames(df)[grepl("sub_",colnames(df))],"cov_bin_history_any_dementia")] <- NULL 
     
+
     check_vitals(df)
     readr::write_rds(df, file.path("output", paste0("model_input-",active_analyses$name[i],".rds")), compress = "gz")
     print(paste0("Saved: output/model_input-",active_analyses$name[i],".rds"))
     rm(df)
-    
+
   }
-  
-  # Make model input: sub_male_age_80_110 ----------------------------------------
-  
-  if (active_analyses$analysis[i]=="sub_male_age_80_110") {
-    
-    print('Make model input: sub_male_age_80_110')
-    
-    df <- input[input$sub_bin_covid19_confirmed_history==FALSE & 
-                  input$cov_cat_sex=="Male" & 
-                  input$cov_num_age>=80 &
-                  input$cov_num_age<111,]
-    
-    df[,c(colnames(df)[grepl("sub_",colnames(df))],"cov_cat_sex")] <- NULL
+
+  # Make model input: prior parkison false -------------------------------------
+
+  if (active_analyses$analysis[i]=="sub_prior_parkinson_true") {
+
+    print('Make model input: sub_prior_parkinson_false')
+
+    df <- input[input$sub_bin_covid19_confirmed_history==FALSE &
```

Comment:
This subgroup should be defined by history of Parkinson's disease rather than history of any dementia.